### DISCUSSION TITLE
What should we do with the errors returned by the action?

### DISCUSSION THREAD

**Discussion by vedantroy** [airjp73/rvf#94] [Q&A] [answered]
I have:
```javascript
    const fieldValues = await validator.validate(await request.formData());
    if (fieldValues.error) return validationError(fieldValues.error);
```
in my action code.
Do I need to do anything with the validation error? (i.e use `useActionData`)?

I notice in the example code that we don't pass in the error to the form component.
I guess it makes sense that we can ignore the errors from the action since validation / displaying of errors is done in the client anyways, but I just wanted to confirm.

**Comment by airjp73** [comment, marked as answer]
Hi! The errors returned from the action are automatically picked up by the form and provided via `useField` just like client-side validation errors would be.

**Reply by vedantroy** [reply]
That's good to know! Is the form determining the route that the POST goes to and using `useFetcher` underneath the hood or something?

**Reply by airjp73** [reply]
Nope, it's just using remix's own `Form` component. So whichever route remix would use normally is the route we use. If you provide a fetcher, then it will use `fetcher.Form`.
